Address mixup related to time in x2sys_datalist

PaulWessel opened the pull request:
See #6170 for background.  The key issue was a mismatch between the parameter "time" which is used in correction tables that need a temporal drift correction, and the actual time data column called "rtime".  Because time may be passed into GMT as rtime, atime or ytime, we need to check for this and only compare against "time".  With this fix the rest of the code example seems to work.  I added a long comment to explain why the extra checks.

I also spent time tracking down the scary

`x2sys_datalist (x2sys.c:1097(x2sys_free_list)): tried to free unallocated memory`

one gets under **-Vd** but those were only meant to inform developers that _gmt_M_free_ was given a NULL pointer - there is no problem here.  However, I changed the message to something more benign:

`x2sys_datalist (mgd77.c:4295(MGD77_end)): FYI: gmt_M_free given a NULL pointer - ignored`

as well as preventing two functions freeing arrays of lists not to bother if the number of entries is zero.

Hopefully @bjcoakley can give this branch a spin unless it gets merger before then,
 Closes #6170.

bjcoakley commented:
Paul

Have these changes been incorporated into the “latest stable release” or the “Current development version?”

Plans to?

THanks

Bernie


> On Apr 4, 2022, at 1:51 AM, Paul Wessel ***@***.***> wrote:
> 
> See #6170 <https://github.com/GenericMappingTools/gmt/issues/6170> for background. The key issue was a mismatch between the parameter "time" which is used in correction tables that need a temporal drift correction, and the actual time data column called "rtime". Because time may be passed into GMT as rtime, atime or ytime, we need to check for this and only compare against "time". With this fix the rest of the code example seems to work. I added a long comment to explain why the extra checks.
> I also spent time tracking down the scary
> 
> x2sys_datalist (x2sys.c:1097(x2sys_free_list)): tried to free unallocated memory
> 
> one gets under -Vd but those were only meant to inform developers that gmt_M_free was given a NULL pointer - there is no problem here. However, I changed the message to something more benign:
> 
> x2sys_datalist (mgd77.c:4295(MGD77_end)): FYI: gmt_M_free given a NULL pointer - ignored
> 
> as well as preventing two functions freeing arrays of lists not to bother if the number of entries is zero.
> 
> Hopefully @bjcoakley <https://github.com/bjcoakley> can give this branch a spin unless it gets merger before then,
> Closes #6170 <https://github.com/GenericMappingTools/gmt/issues/6170>.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/GenericMappingTools/gmt/pull/6514 <https://github.com/GenericMappingTools/gmt/pull/6514>
> Commit Summary
> 
> f516bf5 <https://github.com/GenericMappingTools/gmt/pull/6514/commits/f516bf5b40cdec625decc4009c21a366a64fe760> Address mixup related to time in x2sys_datalist
> File Changes (3 files <https://github.com/GenericMappingTools/gmt/pull/6514/files>)
> M src/gmt_memory.c <https://github.com/GenericMappingTools/gmt/pull/6514/files#diff-c3a715b1dd40c429a0ff54d4e54285234004e14379b22a4568fbbaf8aa33a3af> (2)
> M src/gmt_support.c <https://github.com/GenericMappingTools/gmt/pull/6514/files#diff-2508d95f3a6c81a65f8779495bdab181d7b3bfd239cbbc2d0f4c3c6c628c0a4b> (1)
> M src/x2sys/x2sys.c <https://github.com/GenericMappingTools/gmt/pull/6514/files#diff-4a159c7434da9ba70a6535477f74fdf700c19e739cde8c5827322e2a1456290e> (21)
> Patch Links:
> 
> https://github.com/GenericMappingTools/gmt/pull/6514.patch <https://github.com/GenericMappingTools/gmt/pull/6514.patch>
> https://github.com/GenericMappingTools/gmt/pull/6514.diff <https://github.com/GenericMappingTools/gmt/pull/6514.diff>
> —
> Reply to this email directly, view it on GitHub <https://github.com/GenericMappingTools/gmt/pull/6514>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALTHMJU3HG3PIZSHMPLRUQTVDK3TVANCNFSM5SO6X47A>.
> You are receiving this because you were mentioned.
>

PaulWessel commented:
Hi Bernie-

They were merged into master repo yesterday so if you can build from git
then you have access.
Otherwise 6.4 will happen in ~1 month.

Paul Wessel
Professor, Department of Earth Sciences
(808) 956-4778 | ***@***.***
http://www.soest.hawaii.edu/earthsciences

On April 5, 2022 at 12:43:09 AM, bjcoakley ***@***.***) wrote:

Paul

Have these changes been incorporated into the “latest stable release” or
the “Current development version?”

Plans to?

THanks

Bernie


> On Apr 4, 2022, at 1:51 AM, Paul Wessel ***@***.***> wrote:
>
> See #6170 <https://github.com/GenericMappingTools/gmt/issues/6170> for
background. The key issue was a mismatch between the parameter "time" which
is used in correction tables that need a temporal drift correction, and the
actual time data column called "rtime". Because time may be passed into GMT
as rtime, atime or ytime, we need to check for this and only compare
against "time". With this fix the rest of the code example seems to work. I
added a long comment to explain why the extra checks.
> I also spent time tracking down the scary
>
> x2sys_datalist (x2sys.c:1097(x2sys_free_list)): tried to free unallocated
memory
>
> one gets under -Vd but those were only meant to inform developers that
gmt_M_free was given a NULL pointer - there is no problem here. However, I
changed the message to something more benign:
>
> x2sys_datalist (mgd77.c:4295(MGD77_end)): FYI: gmt_M_free given a NULL
pointer - ignored
>
> as well as preventing two functions freeing arrays of lists not to bother
if the number of entries is zero.
>
> Hopefully @bjcoakley <https://github.com/bjcoakley> can give this branch
a spin unless it gets merger before then,
> Closes #6170 <https://github.com/GenericMappingTools/gmt/issues/6170>.
>
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/GenericMappingTools/gmt/pull/6514 <
https://github.com/GenericMappingTools/gmt/pull/6514>
> Commit Summary
>
> f516bf5 <
https://github.com/GenericMappingTools/gmt/pull/6514/commits/f516bf5b40cdec625decc4009c21a366a64fe760>
Address mixup related to time in x2sys_datalist
> File Changes (3 files <
https://github.com/GenericMappingTools/gmt/pull/6514/files>)
> M src/gmt_memory.c <
https://github.com/GenericMappingTools/gmt/pull/6514/files#diff-c3a715b1dd40c429a0ff54d4e54285234004e14379b22a4568fbbaf8aa33a3af>
(2)
> M src/gmt_support.c <
https://github.com/GenericMappingTools/gmt/pull/6514/files#diff-2508d95f3a6c81a65f8779495bdab181d7b3bfd239cbbc2d0f4c3c6c628c0a4b>
(1)
> M src/x2sys/x2sys.c <
https://github.com/GenericMappingTools/gmt/pull/6514/files#diff-4a159c7434da9ba70a6535477f74fdf700c19e739cde8c5827322e2a1456290e>
(21)
> Patch Links:
>
> https://github.com/GenericMappingTools/gmt/pull/6514.patch <
https://github.com/GenericMappingTools/gmt/pull/6514.patch>
> https://github.com/GenericMappingTools/gmt/pull/6514.diff <
https://github.com/GenericMappingTools/gmt/pull/6514.diff>
> —
> Reply to this email directly, view it on GitHub <
https://github.com/GenericMappingTools/gmt/pull/6514>, or unsubscribe <
https://github.com/notifications/unsubscribe-auth/ALTHMJU3HG3PIZSHMPLRUQTVDK3TVANCNFSM5SO6X47A
>.
> You are receiving this because you were mentioned.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/GenericMappingTools/gmt/pull/6514#issuecomment-1088126015>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AGJ7IX6UNWW77FPFYIJJ7ULVDN5ATANCNFSM5SO6X47A>
.
You are receiving this because you modified the open/close state.Message
ID: ***@***.***>
